<a href="https://colab.research.google.com/github/djbell01/563-DawsonBell/blob/main/Problem_Set_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np

$1.1$

Consider a system $A$ consisting of subsystems $A_{1}$ and $A_{2}$, for which $\Omega_{1} = 10^{20}$ and $\Omega_{2} = 10^{22}$. What is the number of configurations available to the combined system? Also, compute the entropies $S$, $S_{1}$, and $S_{2}$.


$$Ω_{combined} = Ω_{1} * Ω_{2}$$

$$S = k_{B}ln(Ω)$$

In [ ]:
omega_1 = 10e20
omega_2 = 10e22
k_B = 1.381e-23 # units of J/K

combined_states = omega_1*omega_2

S_1 = k_B*np.log(omega_1)
S_2 = k_B*np.log(omega_2)
S = k_B*np.log(combined_states)

print(f'Combined number of states = {combined_states}')
print(f'Entropy of A_1 = {S_1} J/K')
print(f'Entropy of A_2 = {S_2} J/K')
print(f'Combined system entropy = {S} J/K')

Combined number of states = 9.999999999999999e+43
Entropy of A_1 = 6.677727028192031e-22 J/K
Entropy of A_2 = 7.3137010308769875e-22 J/K
Combined system entropy = 1.3991428059069019e-21 J/K


$1.2$

By what factor does the number of available configurations increase when $10\ m^{3}$ of air at $1.0\ atm$ and $300\ K$ is allowed to expand by $0.001\%$ at constant temperature?

$$S = Nk_{B}ln\biggl(\frac{Ve^{5/2}}{Nλ_{th}^{3}}\biggr)$$

$$S = k_{B}ln(Ω)$$

$$k_{B}ln(Ω) = Nk_{B}ln\biggl(\frac{Ve^{5/2}}{Nλ_{th}^{3}}\biggr)$$

$$e^{ln(Ω)} = e^{Nln(\frac{Ve^{5/2}}{Nλ_{th}^{3}})}$$

$$Ω = e^{ln((\frac{Ve^{5/2}}{Nλ_{th}^{3}})^{N})}$$

$$Ω = \biggl(\frac{Ve^{5/2}}{Nλ_{th}^{3}}\biggl)^{N}$$

$$\frac{Ω_{2}}{Ω_{1}} = \frac{\biggl(\frac{V_{2}e^{5/2}}{Nλ_{th}^{3}}\biggl)^{N}}{\biggl(\frac{V_{1}e^{5/2}}{Nλ_{th}^{3}}\biggl)^{N}}$$

$$\frac{Ω_{2}}{Ω_{1}} = \Bigg(\frac{\frac{V_{2}e^{5/2}}{Nλ_{th}^{3}}}{\frac{V_{1}e^{5/2}}{Nλ_{th}^{3}}}\Bigg)^{N}$$

$$\frac{Ω_{2}}{Ω_{1}} = \bigg(\frac{V_{2}}{V_{1}}\bigg)^{N}$$

In [9]:
# n = 100000000 # any number, because we only care about the factor
# n_2 = n*1.00001
# n_cubed = n**3
# n_2_cubed = n_2**3
# ratio = n_2_cubed/n_cubed
# print(f'Factor = {ratio}')

P = 101325 # Pa
T = 300 # K
V_1 = 10 # m^3
k_B = 1.38065E-23 # J/K

V_2 = V_1*1.00001 # m^3
N = (P*V_1)/(k_B*T)


fraction = (V_2/V_1)
print(f'Factor: {fraction}^{N}')

Factor: 1.00001^2.446311519936262e+26


$1.3$

By what factor does the number of available configurations increase when $150\ kJ$ is added to a system containing $2.0\ mol$ of particles at constant volume and $T = 300\ K$?

$$E = \frac{3}{2}RT$$

In [ ]:
E = 150 #kJ
R = 0.008314 #kJ/molK
T_initial = 300 #K
n = 2 #mol

T_final = T_initial + (2/3)*E/R
dSdE = 1/T_final

$1.4$

A sample consisting of five molecules has a total energy $5ϵ$. Each molecule is able to occupy states of energy $ϵj$, with $j = 0,1,2,...,∞$. Draw up a table with columns by the energy of the states and write beneath them all configurations that are consistent with the total energy. Identify the type of configuration that is most probable.